src/components/sharedComponents/Sidebar.js

```diff
@@ -0,0 +1,50 @@
+
+import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import '../../scss/sidebar.scss';
+
+class SideBar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      slide: true,
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange() {
+    this.setState((prevState) => ({
+      slide: !prevState.slide,
+    }));
+  }
+
+  render() {
+    const { slide } = this.state;
+    const style = {
+      display: slide ? 'block' : 'none',
+    };
+    const buttonStyle = {
+      left: slide ? '194px' : '2px',
+    };
+    return (
+      <div className="wrapper">
+        <div className="sidebar" style={style}>
+          <Route>
+            {this.props.children}
+          </Route>
+        </div>
+        <button type="button" style={buttonStyle} onClick={this.handleChange}>{slide ? 'close' : 'open'}</button>
+      </div>
+    );
```

Comment:
You could add a translation for these (`close` and `open`) too!

Reply:
I added the translation, check out the Heroku app to test out the changes